Tools/autotest/pysim/util.py

```diff
@@ -197,7 +199,13 @@ def start_SITL(binary, valgrind=False, gdb=False, wipe=False, synthetic_clock=Tr
         f.write("r\n")
         f.close()
         cmd.extend(['xterm', '-e', 'gdb', '-x', '/tmp/x.gdb', '--args'])
-    
+    if gdbserver:
+        f = open("/tmp/x.gdb", "w")
+        f.write("target extended-remote localhost:3333\nc\n")
+        f.close()
+        cmd.extend(['gdbserver', 'localhost:3333'])
+        run_cmd('screen -d -m -S ardupilot-gdb bash -c "gdb -x /tmp/x.gdb"')
```

Comment:
Using screen is nice, and should definitely be an option.

However, having this option work differently to the `--gdb` option is not good.

In fact, we should probably look at using "run-in-terminal-window.sh" for both of these.